### ISSUE TITLE
Using kata with kubeadm causes crashloop

### ISSUE THREAD

**Issue by jamiehannaford** [kata-containers/runtime#341]
# Description of problem

- I installed Kata to Ubuntu 16.04 using [these instructions](https://github.com/kata-containers/documentation/blob/master/install/ubuntu-installation-guide.md). 
- I then installed `kubeadm`, `kubelet` and `kubectl` using [these instructions](https://kubernetes.io/docs/tasks/tools/install-kubeadm/#installing-kubeadm-kubelet-and-kubectl).
- I ran `sysctl net.bridge.bridge-nf-call-iptables=1` (required for most CNI providers)
- I finally ran `kubeadm init`

# Expected result

```
[init] Using Kubernetes version: v1.10.3
[init] Using Authorization modes: [Node RBAC]
[preflight] Running pre-flight checks.
	[WARNING FileExisting-crictl]: crictl not found in system path
Suggestion: go get github.com/kubernetes-incubator/cri-tools/cmd/crictl
[certificates] Generated ca certificate and key.
[certificates] Generated apiserver certificate and key.
[certificates] apiserver serving cert is signed for DNS names [nested-vm-3 kubernetes kubernetes.default kubernetes.default.svc kubernetes.default.svc.cluster.local] and IPs [10.96.0.1 10.142.0.5]
[certificates] Generated apiserver-kubelet-client certificate and key.
[certificates] Generated etcd/ca certificate and key.
[certificates] Generated etcd/server certificate and key.
[certificates] etcd/server serving cert is signed for DNS names [localhost] and IPs [127.0.0.1]
[certificates] Generated etcd/peer certificate and key.
[certificates] etcd/peer serving cert is signed for DNS names [nested-vm-3] and IPs [10.142.0.5]
[certificates] Generated etcd/healthcheck-client certificate and key.
[certificates] Generated apiserver-etcd-client certificate and key.
[certificates] Generated sa key and public key.
[certificates] Generated front-proxy-ca certificate and key.
[certificates] Generated front-proxy-client certificate and key.
[certificates] Valid certificates and keys now exist in "/etc/kubernetes/pki"
[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/admin.conf"
[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/kubelet.conf"
[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/controller-manager.conf"
[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/scheduler.conf"
[controlplane] Wrote Static Pod manifest for component kube-apiserver to "/etc/kubernetes/manifests/kube-apiserver.yaml"
[controlplane] Wrote Static Pod manifest for component kube-controller-manager to "/etc/kubernetes/manifests/kube-controller-manager.yaml"
[controlplane] Wrote Static Pod manifest for component kube-scheduler to "/etc/kubernetes/manifests/kube-scheduler.yaml"
[etcd] Wrote Static Pod manifest for a local etcd instance to "/etc/kubernetes/manifests/etcd.yaml"
[init] Waiting for the kubelet to boot up the control plane as Static Pods from directory "/etc/kubernetes/manifests".
[init] This might take a minute or longer if the control plane images have to be pulled.
[apiclient] All control plane components are healthy after 41.501669 seconds
[uploadconfig] Storing the configuration used in ConfigMap "kubeadm-config" in the "kube-system" Namespace
[markmaster] Will mark node nested-vm-3 as master by adding a label and a taint
[markmaster] Master nested-vm-3 tainted and labelled with key/value: node-role.kubernetes.io/master=""
[bootstraptoken] Using token: jnima9.1vepq63h0f2e6e4g
[bootstraptoken] Configured RBAC rules to allow Node Bootstrap tokens to post CSRs in order for nodes to get long term certificate credentials
[bootstraptoken] Configured RBAC rules to allow the csrapprover controller automatically approve CSRs from a Node Bootstrap Token
[bootstraptoken] Configured RBAC rules to allow certificate rotation for all node client certificates in the cluster
[bootstraptoken] Creating the "cluster-info" ConfigMap in the "kube-public" namespace
[addons] Applied essential addon: kube-dns
[addons] Applied essential addon: kube-proxy

Your Kubernetes master has initialized successfully!

To start using your cluster, you need to run the following as a regular user:

  mkdir -p $HOME/.kube
  sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
  sudo chown $(id -u):$(id -g) $HOME/.kube/config

You should now deploy a pod network to the cluster.
Run "kubectl apply -f [podnetwork].yaml" with one of the options listed at:
  https://kubernetes.io/docs/concepts/cluster-administration/addons/

You can now join any number of machines by running the following on each node
as root:

  kubeadm join <ipaddress>:6443 --token jnima9.1vepq63h0f2e6e4g --discovery-token-ca-cert-hash sha256:d273d448d82eaca6522466e176dc7a63421bc5190401f77103572b2f867bc59c
```

# Actual result

```
[init] Using Kubernetes version: v1.10.3
[init] Using Authorization modes: [Node RBAC]
[preflight] Running pre-flight checks.
	[WARNING FileExisting-crictl]: crictl not found in system path
Suggestion: go get github.com/kubernetes-incubator/cri-tools/cmd/crictl
[certificates] Generated ca certificate and key.
[certificates] Generated apiserver certificate and key.
[certificates] apiserver serving cert is signed for DNS names [nested-vm-3 kubernetes kubernetes.default kubernetes.default.svc kubernetes.default.svc.cluster.local] and IPs [10.96.0.1 10.142.0.5]
[certificates] Generated apiserver-kubelet-client certificate and key.
[certificates] Generated etcd/ca certificate and key.
[certificates] Generated etcd/server certificate and key.
[certificates] etcd/server serving cert is signed for DNS names [localhost] and IPs [127.0.0.1]
[certificates] Generated etcd/peer certificate and key.
[certificates] etcd/peer serving cert is signed for DNS names [nested-vm-3] and IPs [10.142.0.5]
[certificates] Generated etcd/healthcheck-client certificate and key.
[certificates] Generated apiserver-etcd-client certificate and key.
[certificates] Generated sa key and public key.
[certificates] Generated front-proxy-ca certificate and key.
[certificates] Generated front-proxy-client certificate and key.
[certificates] Valid certificates and keys now exist in "/etc/kubernetes/pki"
[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/admin.conf"
[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/kubelet.conf"
[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/controller-manager.conf"
[kubeconfig] Wrote KubeConfig file to disk: "/etc/kubernetes/scheduler.conf"
[controlplane] Wrote Static Pod manifest for component kube-apiserver to "/etc/kubernetes/manifests/kube-apiserver.yaml"
[controlplane] Wrote Static Pod manifest for component kube-controller-manager to "/etc/kubernetes/manifests/kube-controller-manager.yaml"
[controlplane] Wrote Static Pod manifest for component kube-scheduler to "/etc/kubernetes/manifests/kube-scheduler.yaml"
[etcd] Wrote Static Pod manifest for a local etcd instance to "/etc/kubernetes/manifests/etcd.yaml"
[init] Waiting for the kubelet to boot up the control plane as Static Pods from directory "/etc/kubernetes/manifests".
[init] This might take a minute or longer if the control plane images have to be pulled.
```

It then hangs at this point and the SSH connection is terminated (timed out). For some reason SSH becomes unresponsive after that. At first I thought it was a kernel panic, but I was able to retrieve some logs after this point via the GCE control panel:

```
May 25 20:50:46 nested-vm-2 kata-runtime[2728]: time="2018-05-25T20:50:46.274423721Z" level=info arguments="\"create --bundle /var/run/docker/containerd/daemon/io.containerd.runtime.v1.linux/moby/2e0cd473a23ff2a1dab613f06ee4574e48aa1b07d30ab3d3d92313fc14ebb1f2 --pid-file /run/docker/containerd/daemon/io.containerd.runtime.v1.linux/moby/2e0cd473a23ff2a1dab613f06ee4574e48aa1b07d30ab3d3d92313fc14ebb1f2/init.pid 2e0cd473a23ff2a1dab613f06ee4574e48aa1b07d30ab3d3d92313fc14ebb1f2\"" command=create commit= name=kata-runtime pid=2728 source=runtime version=1.0.0
May 25 20:50:46 nested-vm-2 kata-runtime[2728]: time="2018-05-25T20:50:46.276924473Z" level=warning msg="shortening QMP socket name" arch=amd64 name=kata-runtime new-name=mon-3b99f2da-c93a-47bf-a485-2e original-name=mon-3b99f2da-c93a-47bf-a485-2e34394efa9d pid=2728 source=virtcontainers subsystem=qemu
May 25 20:50:46 nested-vm-2 kata-runtime[2728]: time="2018-05-25T20:50:46.277054533Z" level=warning msg="shortening QMP socket name" arch=amd64 name=kata-runtime new-name=ctl-3b99f2da-c93a-47bf-a485-2e original-name=ctl-3b99f2da-c93a-47bf-a485-2e34394efa9d pid=2728 source=virtcontainers subsystem=qemu
May 25 20:50:46 nested-vm-2 kata-runtime[2728]: time="2018-05-25T20:50:46.277711512Z" level=info msg="Attaching virtual endpoint" arch=amd64 name=kata-runtime pid=2728 source=virtcontainers subsystem=network
May 25 20:50:46 nested-vm-2 kata-runtime[2728]: time="2018-05-25T20:50:46.277815087Z" level=error msg="Error bridging virtual ep" arch=amd64 error="Could not get veth interface: docker0: Incorrect link type bridge, expecting veth" name=kata-runtime pid=2728 source=virtcontainers subsystem=network
May 25 20:50:46 nested-vm-2 kata-runtime[2728]: time="2018-05-25T20:50:46.277852303Z" level=error msg="Could not get veth interface: docker0: Incorrect link type bridge, expecting veth" command=create name=kata-runtime pid=2728 source=runtime
May 25 20:50:46 nested-vm-2 dockerd[1251]: time="2018-05-25T20:50:46Z" level=debug msg="event published" module="containerd/containers" ns=moby topic="/containers/delete" type=containerd.events.ContainerDelete
May 25 20:50:46 nested-vm-2 dockerd[1251]: time="2018-05-25T20:50:46Z" level=info msg="shim reaped" id=2e0cd473a23ff2a1dab613f06ee4574e48aa1b07d30ab3d3d92313fc14ebb1f2 module="containerd/tasks"
May 25 20:50:46 nested-vm-2 dockerd[1251]: time="2018-05-25T20:50:46.288574552Z" level=debug msg="EnableService a67e150d2e074c4314b354c3ab4a59543721ca6721dd4871891b55890f9be156 START"
May 25 20:50:46 nested-vm-2 dockerd[1251]: time="2018-05-25T20:50:46.288662771Z" level=debug msg="EnableService a67e150d2e074c4314b354c3ab4a59543721ca6721dd4871891b55890f9be156 DONE"
May 25 20:50:46 nested-vm-2 dockerd[1251]: time="2018-05-25T20:50:46.293682328Z" level=error msg="stream copy error: reading from a closed fifo"
May 25 20:50:46 nested-vm-2 dockerd[1251]: time="2018-05-25T20:50:46.293753886Z" level=error msg="stream copy error: reading from a closed fifo"
May 25 20:50:46 nested-vm-2 dockerd[1251]: time="2018-05-25T20:50:46Z" level=debug msg="event published" module="containerd/containers" ns=moby topic="/containers/delete" type=containerd.events.ContainerDelete
May 25 20:50:46 nested-vm-2 dockerd[1251]: time="2018-05-25T20:50:46.305556369Z" level=error msg="9631d785ad34c971ebeb45a2ea2f0c8da641e3fcc0d7c24a0986680f56a623fc cleanup: failed to delete container from containerd: no such container"
May 25 20:50:46 nested-vm-2 dockerd[1251]: time="2018-05-25T20:50:46.306565334Z" level=error msg="Handler for POST /v1.31/containers/9631d785ad34c971ebeb45a2ea2f0c8da641e3fcc0d7c24a0986680f56a623fc/start returned error: OCI runtime create failed: Could not get veth interface: docker0: Incorrect link type bridge, expecting veth: unknown"
```

[Here](https://gist.github.com/jamiehannaford/210f463cab0e88b3f049d2d29a1aa686) are full logs.

**Comment by jamiehannaford** [comment]
Whoops, this is fixed by _actually_ reading the [docs](https://github.com/kata-containers/documentation/blob/master/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md) 😂

**Comment by ygefen** [comment]
Hey @jamiehannaford, would you please be able to elaborate on the cause of your issue and how you fixed it?

**Comment by jamesodhunt** [comment]
Hi - this issue is labelled `release-gating` but doesn't have an associated PR.

Please can you comment on the status of this item so we know if it could be
included in the [upcoming 2.0 release](https://github.com/kata-containers/kata-containers/wiki/Kata-Containers-2.0-Roadmap)?
